src/project.coffee

```diff
@@ -30,8 +30,8 @@ class Project extends Model
     @consumeServices(packageManager)
 
   destroyed: ->
-    buffer.destroy() for buffer in @buffers
-    repository?.destroy() for repository in @repositories
+    buffer.destroy() for buffer in @buffers.slice()
+    repository?.destroy() for repository in @repositories.slice()
```

Comment:
At some point I got an error (calling destroy on undefined). Something in an event handler was mutating the array being iterated over and, since these CS loops compile to plain old for loops, it would end up accessing bad indexes.

I'm not sure why the existential operator in in the second line but I kinda wondered if it wasn't to work around the same problem and figured, even if it wasn't, it's good not to iterate over a thing you might muck with so I changed that too XD